### PR TITLE
Adapter raw data endpoint

### PR DESCRIPTION
**Pull request by mathiaszinnen**

### CODE REVIEW

File: adapter/src/main/java/org/jvalue/ods/adapterservice/adapter/api/rest/v1/AdapterEndpoint.java [hunk 1]
```diff
@@ -16,32 +16,58 @@
 @RequestMapping("/")
 public class AdapterEndpoint {
 
-  private final Adapter adapter;
+    private final Adapter adapter;
```
Comment (sonallux) [inline adapter/src/main/java/org/jvalue/ods/adapterservice/adapter/api/rest/v1/AdapterEndpoint.java]: Why did you change the indentation to 4 spaces? According to the `.editorconfig` file in the root of the repository, all files should be intended with two spaces.

**Review by sonallux — CHANGES_REQUESTED** [review]

File: adapter/src/main/java/org/jvalue/ods/adapterservice/adapter/api/rest/v1/AdapterEndpoint.java (hunk 1 above)
Comment (mathiaszinnen) [reply adapter/src/main/java/org/jvalue/ods/adapterservice/adapter/api/rest/v1/AdapterEndpoint.java]: I blame IntelliJ!
Thanks for pointing it out, I will fix that

File: adapter/README.md [hunk 2]
```diff
@@ -50,20 +52,25 @@ When started via docker-compose *base_url* is `http://localhost:9000/api/adapter
...
-      "type": "HTTP",
-      "parameters": {
-        "location": String,
-        "encoding": String
-      }
-    }
+    "protocol": ProtocolConfig 
     "format": {
-      "type": "JSON" | "XML" | "CSV"
+      "type": "JSON" | "XML" | "CSV" | "RAW"
```
Comment (georg-schwarz) [inline adapter/README.md]: We removed that right?

File: adapter/README.md [hunk 3]
```diff
@@ -130,7 +138,7 @@ When started via docker-compose *base_url* is `http://localhost:9000/api/adapter
 ```
 {
   "format": {
-    "type": "JSON" | "XML" | "CSV"
+    "type": "JSON" | "XML" | "CSV" | "RAW"
```
Comment (georg-schwarz) [inline adapter/README.md]: Same here

File: adapter/src/main/java/org/jvalue/ods/adapterservice/adapter/api/rest/v1/AdapterEndpoint.java [hunk 4]
```diff
@@ -23,23 +23,49 @@ public AdapterEndpoint(Adapter adapter) {
...
+      throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, e.getMessage());
+    }
+  }
+
+  @PostMapping(value = Mappings.RAW_PREVIEW_PATH, produces = "application/json")
+  public ResponseEntity<?> executeRawPreview(
+    @Valid @RequestBody ProtocolConfig config,
+    @RequestParam(required = false, defaultValue = "true") boolean includeData) {
+    try {
+      DataBlob imported = adapter.executeRawImport(config, includeData);
```
Comment (georg-schwarz) [inline adapter/src/main/java/org/jvalue/ods/adapterservice/adapter/api/rest/v1/AdapterEndpoint.java]: 2nd parameter here called `ìncludeData` and in method called `persistData`. Aren't those two different things? Are we mixing things up? Or is it two names for the same thing?

File: adapter/src/main/java/org/jvalue/ods/adapterservice/adapter/interpreter/CsvInterpreter.java [hunk 5]
```diff
@@ -57,12 +57,12 @@ protected void validateParameters(Map<String, Object> inputParameters) {
   }
 
   @Override
-  protected JsonNode doInterpret(String data, Map<String, Object> parameters) throws IOException {
+  protected String doInterpret(String data, Map<String, Object> parameters) throws IOException {
```
Comment (georg-schwarz) [inline adapter/src/main/java/org/jvalue/ods/adapterservice/adapter/interpreter/CsvInterpreter.java]: Can we undo this change since we have no raw interpreter anymore?

File: adapter/src/main/java/org/jvalue/ods/adapterservice/datasource/api/rest/v1/DatasourceEndpoint.java [hunk 6]
```diff
@@ -88,10 +88,8 @@ public void deleteAllDatasources() {
       .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Can not find Datasource with id: " + id));
     try {
       return datasourceManager.trigger(id, runtimeParameters);
-    } catch (IllegalArgumentException e) {
-      throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Invalid datasource or parameters: " + e.getMessage());
-    } catch (RestClientException e) {
-      throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Failed to load data: " + e.getMessage());
+    } catch (ResponseStatusException e) {
```
Comment (georg-schwarz) [inline adapter/src/main/java/org/jvalue/ods/adapterservice/datasource/api/rest/v1/DatasourceEndpoint.java:89]: unnecessary

**Review by georg-schwarz — CHANGES_REQUESTED** [review]

File: adapter/src/main/java/org/jvalue/ods/adapterservice/adapter/api/rest/v1/AdapterEndpoint.java (hunk 4 above)
Comment (mathiaszinnen) [reply adapter/src/main/java/org/jvalue/ods/adapterservice/adapter/api/rest/v1/AdapterEndpoint.java]: IncludeData implies that it does not need to persisted. However, I should probably remove that parameter altogether since in a non-raw import requests with includeData param are persisted as well. 
Maybe we should prevent persistence in both cases, but this should probably be done in another PR, what do you think?

File: adapter/src/main/java/org/jvalue/ods/adapterservice/adapter/interpreter/CsvInterpreter.java (hunk 5 above)
Comment (mathiaszinnen) [reply adapter/src/main/java/org/jvalue/ods/adapterservice/adapter/interpreter/CsvInterpreter.java]: Yeah, good point, I will do that. The JsonNode return type was more precise.

File: adapter/src/main/java/org/jvalue/ods/adapterservice/adapter/api/rest/v1/AdapterEndpoint.java (hunk 4 above)
Comment (georg-schwarz) [reply adapter/src/main/java/org/jvalue/ods/adapterservice/adapter/api/rest/v1/AdapterEndpoint.java]: Yeah move to other PR. Maybe remove that parameter for the raw preview endpoint completly in this PR?

File: adapter/src/main/java/org/jvalue/ods/adapterservice/datasource/api/rest/v1/DatasourceEndpoint.java (hunk 6 above)
Comment (mathiaszinnen) [reply adapter/src/main/java/org/jvalue/ods/adapterservice/datasource/api/rest/v1/DatasourceEndpoint.java:89]: This is actually needed to prevent all response status exceptions from being caught by the subsequent catch clause.

Comment (georg-schwarz) [reply adapter/src/main/java/org/jvalue/ods/adapterservice/datasource/api/rest/v1/DatasourceEndpoint.java:89]: Ok thanks, didn't see that in the diff..

**Review by georg-schwarz — APPROVED** [review]
LGTM
